src/Middleware/HttpLogging/src/RequestBufferingStream.cs

```diff
@@ -7,20 +7,105 @@
 
 namespace Microsoft.AspNetCore.HttpLogging;
 
-internal sealed class RequestBufferingStream : BufferingStream
+internal sealed class RequestBufferingStream : Stream
 {
     private readonly Encoding _encoding;
     private readonly int _limit;
+    private readonly ILogger _logger;
+    private readonly Stream _innerStream;
+    private readonly BufferingWriter _bufferingWriter;
 
     public bool HasLogged { get; private set; }
 
     public RequestBufferingStream(Stream innerStream, int limit, ILogger logger, Encoding encoding)
-        : base(innerStream, logger)
     {
         _logger = logger;
         _limit = limit;
         _innerStream = innerStream;
         _encoding = encoding;
+        _bufferingWriter = new BufferingWriter();
+    }
+
+    public override bool CanSeek => _innerStream.CanSeek;
```

Comment:
Most of the changes to this file & `ResponseBufferingStream` are moving the `Stream` overrides from `BufferingStream` (now `BufferingWriter`) to these 2 files which now inherit from `Stream`